src/app/modules/add-event/components/add-event/add-event.component.ts

```diff
@@ -227,7 +227,7 @@ export class AddEventComponent implements OnInit {
   }
 
   private generateEventBodyByRouteAndUserId(route: Route, userId: number): BaseEventDto | BicycleCompetitionDto {
-    const eventType: EventType | null = detectEventTypeBySportType(BaseEventType.workout, this.eventForm.get('sportType')?.value)
+    const eventType: EventType | null = detectEventTypeBySportType(this.eventForm.get('eventType')?.value, this.eventForm.get('sportType')?.value)
```

Comment:
```suggestion
    const eventType: EventType | null = detectEventTypeBySportType(this.eventForm.value.eventType, this.eventForm.value.sportType)
```
Тоже самое, только без optional-chaning-а. Да и код читаемее

Reply:
Правда это будет работать только в том, случае если эти контролы не деактивированы через `.disable()`

Reply:
Да. Так лучше. Ща поменяю.
Эта форма никогда не будет задизейблена.